restructuredtext_lint/lint.py

```diff
@@ -80,9 +75,24 @@ def error_collector(data):
     return errors
 
 
-def lint_file(filepath, encoding=None):
+def lint(content, filepath=None, directives=None, roles=None):
+    """Lint reStructuredText and return errors
+
+    :param string content: reStructuredText to be linted
+    :param string filepath: Optional path to file, this will be returned as the source
+    :param list directives: custom directives to ignore
+    :param list roles: custom roles to ignore
+    :rtype list: List of errors. Each error will contain a line, source (filepath),
+        message (error message), and full message (error message + source lines)
+    """
+    with ignore.register_unregister_ignores(directives=directives,
```

Comment:
No, this PR is solving both problems at once. I thought it was clear that was the direction we were taking. Having custom doesn't make sense otherwise...